In [127]:
# import necessary packages
import numpy as np
import pandas as pd

Function definition

In [128]:
# Calculate measures of central tendency
def central_tendency(col):
    print(f"Mean: {col.mean():.2f}")
    print(f"Median: {col.median():.2f}")
    print(f"Mode: {col.mode().values[0]:.2f}")
    return

# Calculate measures of spread
def measures_of_spread(col):
    print(f"Mean: {col.mean():.2f}")
    print(f"Median: {col.median():.2f}")
    print(f"Mode: {col.mode().values[0]:.2f}")
    return

Cleaning data import values for nulls, errors and column types

In [129]:
# import dataset
df = pd.read_csv('https://raw.githubusercontent.com/notpeter/crunchbase-data/master/companies.csv')
# df.head()


In [130]:
df['funding_total_usd'] = df['funding_total_usd'].replace('-', np.nan)
df['funding_total_usd'].fillna(0, inplace=True)
df['funding_total_usd'] = df['funding_total_usd'].astype(float)
df.loc[:, 'founded_at'] = df['first_funding_at']
df.dropna(subset=['name'], inplace=True)
# Remove comment to view null data from a specific column
# df[df['country_code'].isnull()]
# Remove comment to review total errors per column
# df.isnull().sum()

### Measuring Central Tendency and Spread

Calculate measures of central tendency and spread for a startup

In [131]:
print("Total Funding Raised:")
central_tendency(df['funding_total_usd'])
measures_of_spread(df['funding_total_usd'])

print("\nNumber of Funding Rounds:")
central_tendency(df['funding_rounds'])
measures_of_spread(df['funding_rounds'])

# print("\nNumber of Funding Rounds:")
# central_tendency

Total Funding Raised:
Mean: 14919146.79
Median: 1000000.00
Mode: 0.00
Mean: 14919146.79
Median: 1000000.00
Mode: 0.00

Number of Funding Rounds:
Mean: 1.73
Median: 1.00
Mode: 1.00
Mean: 1.73
Median: 1.00
Mode: 1.00


For "Total Funding Raised" the average funding raised by startups is $14,919,146 USD with an average of 1.7 rounds of funding per startup but it is highly likely that some startups have received exceptionally large funding amounts, which significantly skews the distribution. The mean is much higher than the median, indicating the presence of high-value outliers because probably few startups have received exceptionally large amounts of funding. The mode proves this too because there are many startups with 0 funding (but can also be due to missing data). The variance and range are very large which indicates a presence of outliers at the upper end of the distribution.

There is a lot less variation in the "Funding rounds" and when observing the difference of measures of central tendency which suggests that the data may be less affected by outliers. The measure of spread also contain moderate levels of variance which indicate that the outliers are not as noticeable.


### Outlier analysis